src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,29 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private static final int TOTAL_THREADS = 2;
 
+    private final List<EventListener> eventListeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executor = Executors.newFixedThreadPool(TOTAL_THREADS);
+
+    public void registerListener(EventListener listener) {
+        eventListeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        eventListeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        eventListeners.forEach(listener -> executor.submit(() -> listener.onEvent(event)));
```

Comment:
exception handling is actually not that a bad idea here, we can wrap `listener.onEvent()` in a try/catch just in case, to not disrupt the work of other listeners